test/parallel/test-http-invalid-urls.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const http = require('http');
+const https = require('https');
+const error = 'Unable to determine the domain name';
+
+assert.throws(() => http.get('www.nodejs.org', assert.fail), error);
+assert.throws(() => http.request('www.nodejs.org', assert.fail), error);
+assert.throws(() => http.get('localhost', assert.fail), error);
+assert.throws(() => http.request('localhost', assert.fail), error);
+assert.throws(() => http.get('127.0.0.1', assert.fail), error);
+assert.throws(() => http.request('127.0.0.1', assert.fail), error);
+assert.throws(() => http.get('http://:80/', assert.fail), error);
+assert.throws(() => http.request('http://:80/', assert.fail), error);
+assert.throws(() => https.get('www.nodejs.org', assert.fail), error);
+assert.throws(() => https.request('www.nodejs.org', assert.fail), error);
+assert.throws(() => https.get('localhost', assert.fail), error);
+assert.throws(() => https.request('localhost', assert.fail), error);
+assert.throws(() => https.get('127.0.0.1', assert.fail), error);
+assert.throws(() => https.require('127.0.0.1', assert.fail), error);
+assert.throws(() => https.get('https://:443/', assert.fail), error);
+assert.throws(() => https.require('https://:443/', assert.fail), error);
```

Comment:
I would rather make a helper function that makes a request and sets up a fail function saying which case failed.

Using `assert.fail` is nice and easy, but the error messages it creates in these situations are confusing.